BAT/BCVariable.h

```diff
@@ -53,10 +53,6 @@ class BCVariable
      * @param unitstring Unit string to be printed for variable. */
     BCVariable(const std::string& name, double lowerlimit, double upperlimit, const std::string& latexname = "", const std::string& unitstring = "");
 
-    /**
-     * Destructor */
-    virtual ~BCVariable() {};
```

Comment:
You need to leave this in. Heed the compiler warnings. See this [question](https://stackoverflow.com/questions/7403883/derived-class-with-non-virtual-destructor/7403943)